src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/OutputFactory.java

```diff
@@ -434,13 +427,41 @@ public Void visitVariable(Declaration.Variable tree, Declaration parent) {
         MemoryLayout treeLayout = tree.layout().orElseThrow();
         if (sizeAvailable) {
             currentBuilder.addVar(fieldName, tree.name(), treeLayout, clazz);
+            Type.Function funcPtr = getAsFunctionPointer(tree.type(), true);
+            if (funcPtr != null) {
+                addFunctionPointerInvoker(funcPtr, fieldName, tree.name());
+            }
         } else {
             warn("Layout size not available for " + fieldName);
         }
 
         return null;
     }
 
+    void addFunctionPointerInvoker(Type.Function funcPtr, String javaName, String nativeName) {
+        FunctionDescriptor descriptor = Type.descriptorFor(funcPtr).orElse(null);
+        if (descriptor == null) {
+            //abort
+            return;
+        }
+
+        //generate functional interface
+        if (funcPtr.varargs() && !funcPtr.argumentTypes().isEmpty()) {
+            warn("varargs in callbacks is not supported: " + funcPtr);
+            return;
+        }
```

Comment:
Seems the same as in `generateFunctionalInterface` ? Might be worth it to unify?